File: ProjectPos.FrontOffice.FrontEnd/src/proxy/interfaces/sales-order-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { Currency } from "../enums/currency";
import { SaleType } from "../enums/sale-type.enum";
import { SalesOrderStatus } from "../enums/sales-order-status";
import { CompanyDto } from "./company-dto";
import { PaymentDto } from "./payment-dto";
import { SalesOrderItemDto } from "./sales-order-item-dto";

export interface SalesOrderDto extends FullAuditedEntityDto<number>{
    saleType?: SaleType;
    price?: number;
    vat?: number;
    priceIncludingVat?: number;
    balance?: number;
    paidAmount?: number;
    currency?: Currency;
    isPaid?: boolean;
    customerId?: number;
    userName?: string;
    status?: SalesOrderStatus;
    customer?: CompanyDto;
    customerName?: string;
    salesOrderItems?: SalesOrderItemDto[];
    payments?: PaymentDto[];
}

File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/SalesOrderDto.cs
using ProjectPos.Data.EntityModels;
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services.DTOs
{
    public class SalesOrderDto : FullAuditedEntityDto<int>
    {
        public SaleType? SaleType { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Price { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? PriceIncludingVat { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Vat { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Balance { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? PaidAmount { get; set; }
        public Currency? Currency { get; set; }
        public SalesOrderStatus? Status { get; set; }
        public bool? IsPaid { get; set; }
        public string? UserName { get; set; }
        public int? CustomerId { get; set; }
        public string? CustomerName { get; set; }

        //dependent entities
        public ICollection<SalesOrderItemDto>? SalesOrderItems { get; set; }
        public ICollection<PaymentDto>? Payments { get; set; }

        public CompanyDto? Customer {  get; set; }
    }
}
